api/views.py

```diff
@@ -429,4 +431,63 @@ def get(self, request, answerpaper_id, format=None):
         answerpaper.status = 'completed'
         answerpaper.save()
         serializer = AnswerPaperSerializer(answerpaper)
-        return Response(serializer.data)
\ No newline at end of file
+        return Response(serializer.data)
+
+
+class CourseCreate(APIView):
+    def post(self, request):
+        username = request.data['creator']
+        user = get_object_or_404(User, username=username)
+        user_id = user.id
+        request.data['creator'] = user_id
+        for learning_module in request.data['learning_module']:
+            learning_module['creator'] = user_id
+            for learning_unit in learning_module["learning_unit"]:
+                if learning_unit['type'] == 'quiz':
+                    learning_unit['quiz']['creator'] = user.id
+                if learning_unit['type'] == 'lesson':
+                    learning_unit['lesson']['creator'] = user.id
+        serializer = CourseSerializer(data=request.data)
+        modules = []
+        for learning_module in request.data['learning_module']:
```

Comment:
You can do the serializer part in the previous for loop only where you are setting creator, no need of adding another for loop.